src/main/java/org/folio/auth/authtokenmodule/storage/ApiTokenStore.java

```diff
@@ -0,0 +1,187 @@
+   * is returned.
+   */
+  public Future<Void> checkTokenNotRevoked(ApiToken apiToken) {
+    UUID tokenId = apiToken.getId();
+
+    log.info("Checking revoked status of {} api token id {}",
+      API_TOKEN_SUFFIX, tokenId);
+
+    String select = "SELECT is_revoked FROM " + tableName(tenant, API_TOKEN_SUFFIX) +
+      "WHERE id=$1";
+    Tuple where = Tuple.of(tokenId);
+
+    return pool.withConnection(conn -> {
+      return getRow(conn, select, where).compose(row -> {
+        Boolean isRevoked = row.getBoolean("is_revoked");
+
+        log.info("Revoked status of {} token id {} is {}", API_TOKEN_SUFFIX, tokenId, isRevoked);
+
+        if (!isRevoked) {
+            return Future.succeededFuture();
+        }
+        return Future.failedFuture("API token revoked");
+      });
+    });
+  }
+
+  /**
+   * Revokes the API token.
+   * @param apiToken The API token to revoke.
+   * @return A failed future if the revoke operation failed. Otherwise a succeeded future
+   * is returned.
+   */
+  public Future<Void> revokeToken(ApiToken apiToken) {
+    UUID tokenId = apiToken.getId();
+    log.info("Revoking API token {}", tokenId);
+
+    String update = "UPDATE " + tableName(tenant, API_TOKEN_SUFFIX) +
+        "SET is_revoked=TRUE WHERE id=$1";
+
+    return pool.preparedQuery(update).execute(Tuple.of(tokenId)).mapEmpty();
+  }
+
+  /**
+   * Gets a list of all the API tokens for a given tenant.
+   * @param tenant The tenant.
+   * @return A list of all the API tokens for a tenant.
+   */
+  public Future<List<ApiToken>> getApiTokensForTenant(String tenant) {
+    String select = "SELECT token FROM " + tableName(tenant, API_TOKEN_SUFFIX);
+    List<ApiToken> tokens = new ArrayList<ApiToken>();
+    return pool.query(select).execute().compose(rows -> {
+      for (Row row : rows) {
+        String tokenString = row.getString("token");
+        try {
+          var apiToken = (ApiToken)Token.parse(tokenString, tokenCreator);
+          tokens.add(apiToken);
+        } catch (TokenValidationException e) {
+          log.error("Unable to parse token for tenant {} after retrieving from storage: {}",
+            tenant, e.getMessage());
+          return Future.failedFuture("Unable to parse token when getting from storage");
+        }
+      }
+      log.info("Retrieved {} token rows for tenant {}", rows.rowCount(), tenant);
+      return Future.succeededFuture(tokens);
+    });
+  }
+
+  // public Future<List<String>> getApiTokensForTenant(String tenant) {
+  //   String select = "SELECT token FROM " + tableName(tenant, API_TOKEN_SUFFIX);
+  //   List<String> tokens = new ArrayList<String>();
+  //   return pool.query(select).execute().compose(rows -> {
+  //     for (Row row : rows) {
+  //       String tokenString = row.getString("token");
+  //       tokens.add(tokenString);
+  //     }
+  //     log.info("Retrieved {} token rows for tenant {}", rows.rowCount(), tenant);
+  //     return Future.succeededFuture(tokens);
+  //   });
+  // }
```

Comment:
This is just me being sloppy. What I was going to do was replace the method with the commented method so that clients get string representations of the tokens rather than token objects. I think clients will want strings and not care at all about objects since the primary purpose of this method will be for clients to get access to the tokens for use in other contexts and users will basically copy paste them.